src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -169,8 +112,8 @@ func (r *ClusterReconciler) Reconcile(
 		}
 	}
 
-	if !reflect.DeepEqual(sts.Status.ReadyReplicas, redpandaCluster.Status.Replicas) {
-		redpandaCluster.Status.Replicas = sts.Status.ReadyReplicas
+	if sts.LastObservedState != nil && !reflect.DeepEqual(sts.LastObservedState.Status.ReadyReplicas, redpandaCluster.Status.Replicas) {
```

Comment:
As the code stands, I think the only way for `LastObservedState` to be `nil` is for `Ensure()` to return an error, right? In that case we'd never reach this line and we wouldn't need the `sts.LastObservedState != nil` check, correct? I don't really mind it, I just want to verify this is the behavior.

Reply:
yes, I added that nil check though, just to be sure. But you're right, if we don't introduce a bug that should never happen